lib/wallet/txdb.js

```diff
@@ -1300,6 +1312,15 @@ class TXDB {
         state.coin(path, 1);
         state.unconfirmed(path, coin.value);
 
+        // FINALIZE is a special case: locked coins _leave_ the wallet.
+        // In this case a TX is erased, adding them back.
+        if (tx.outputs[i] && tx.outputs[i].covenant.type === types.FINALIZE) {
```

Comment:
The `tx.output` and `tx.covenent` helper methods could make this `if` expression a bit cleaner. What do you think about?

```js
if (tx.output(i) && tx.covenant(i).type === types.FINALIZE)
```
The first check on the output protects `tx.covenant` from returning `null`.

https://github.com/handshake-org/hsd/blob/master/lib/primitives/tx.js#L1629